sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/intervalExpressions.scala

```diff
@@ -0,0 +1,107 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.catalyst.expressions.interval
+
+import java.util.Locale
+
+import org.apache.spark.sql.catalyst.expressions.{ExpectsInputTypes, Expression, UnaryExpression}
+import org.apache.spark.sql.catalyst.expressions.codegen.{CodegenContext, ExprCode}
+import org.apache.spark.sql.catalyst.util.IntervalUtils
+import org.apache.spark.sql.catalyst.util.IntervalUtils._
+import org.apache.spark.sql.types._
+import org.apache.spark.unsafe.types.CalendarInterval
+
+abstract class IntervalPart(
+    child: Expression,
+    val dataType: DataType,
+    func: CalendarInterval => Any,
+    funcName: String)
+    extends UnaryExpression
+    with ExpectsInputTypes
+    with Serializable {
+  override def inputTypes: Seq[AbstractDataType] = Seq(CalendarIntervalType)
+  override protected def nullSafeEval(interval: Any): Any = {
+    func(interval.asInstanceOf[CalendarInterval])
+  }
+  override protected def doGenCode(ctx: CodegenContext, ev: ExprCode): ExprCode = {
+    val iu = IntervalUtils.getClass.getName.stripSuffix("$")
+    defineCodeGen(ctx, ev, c => s"$iu.$funcName($c)")
+  }
+}
+
+case class Millenniums(child: Expression)
+    extends IntervalPart(child, IntegerType, getMillenniums, "getMillenniums")
```

Comment:
Oh, it seems that we are confusing the indentation rule on `extends`. We need two spaces instead of four.

Reply:
Please fix all the other instances in this file.

Reply:
Usually, I just check formatting by  `./dev/scalastyle` but this time I decided to try `./dev/scalafmt` because it is new file. Do you think `scalafmt` doesn't conform to the code style?

Reply:
@MaxGekk .
You can see the rule of `Apache Spark` here, https://spark.apache.org/contributing.html, and we follow `https://github.com/databricks/scala-style-guide`.

`scalastyle` and `scalafmt` is just a tool to follow our rule.

Reply:
> we follow https://github.com/databricks/scala-style-guide

Thanks, probably I have found what I will read before bedtime.